hack/testing/rsyslog/viaq_formatting.conf

```diff
@@ -0,0 +1,253 @@
+    }
+    unset $!_EXE;
+    if strlen($!_GID) > 0 then {
+        set $!systemd!t!GID = $!_GID;
+    }
+    unset $!_GID;
+    if strlen($!_HOSTNAME) > 0 then {
+        set $!systemd!t!HOSTNAME = $!_HOSTNAME;
+    }
+    unset $!_HOSTNAME;
+    if strlen($!_PID) > 0 then {
+        set $!systemd!t!PID = $!_PID;
+    }
+    unset $!_PID;
+    if strlen($!_SELINUX_CONTEXT) > 0 then {
+        set $!systemd!t!SELINUX_CONTEXT = $!_SELINUX_CONTEXT;
+    }
+    unset $!_SELINUX_CONTEXT;
+    if strlen($!_SOURCE_REALTIME_TIMESTAMP) > 0 then {
+        set $!systemd!t!SOURCE_REALTIME_TIMESTAMP = $!_SOURCE_REALTIME_TIMESTAMP;
+    }
+    unset $!_SOURCE_REALTIME_TIMESTAMP;
+    if strlen($!_SYSTEMD_CGROUP) > 0 then {
+        set $!systemd!t!SYSTEMD_CGROUP = $!_SYSTEMD_CGROUP;
+    }
+    unset $!_SYSTEMD_CGROUP;
+    if strlen($!_SYSTEMD_OWNER_UID) > 0 then {
+        set $!systemd!t!SYSTEMD_OWNER_UID = $!_SYSTEMD_OWNER_UID;
+    }
+    unset $!_SYSTEMD_OWNER_UID;
+    if strlen($!_SYSTEMD_SESSION) > 0 then {
+        set $!systemd!t!SYSTEMD_SESSION = $!_SYSTEMD_SESSION;
+    }
+    unset $!_SYSTEMD_SESSION;
+    if strlen($!_SYSTEMD_SLICE) > 0 then {
+        set $!systemd!t!SYSTEMD_SLICE = $!_SYSTEMD_SLICE;
+    }
+    unset $!_SYSTEMD_SLICE;
+    if strlen($!_SYSTEMD_UNIT) > 0 then {
+        set $!systemd!t!SYSTEMD_UNIT = $!_SYSTEMD_UNIT;
+    }
+    unset $!_SYSTEMD_UNIT;
+    if strlen($!_SYSTEMD_USER_UNIT) > 0 then {
+        set $!systemd!t!SYSTEMD_USER_UNIT = $!_SYSTEMD_USER_UNIT;
+    }
+    unset $!_SYSTEMD_USER_UNIT;
+    if strlen($!_TRANSPORT) > 0 then {
+        set $!systemd!t!TRANSPORT = $!_TRANSPORT;
+    }
+    unset $!_TRANSPORT;
+    if strlen($!_UID) > 0 then {
+        set $!systemd!t!UID = $!_UID;
+    }
+    unset $!_UID;
+    if strlen($!_KERNEL_DEVICE) > 0 then {
+        set $!systemd!k!KERNEL_DEVICE = $!_KERNEL_DEVICE;
+    }
+    unset $!_KERNEL_DEVICE;
+    if strlen($!_KERNEL_SUBSYSTEM) > 0 then {
+        set $!systemd!k!KERNEL_SUBSYSTEM = $!_KERNEL_SUBSYSTEM;
+    }
+    unset $!_KERNEL_SUBSYSTEM;
+    if strlen($!_UDEV_SYSNAME) > 0 then {
+        set $!systemd!k!UDEV_SYSNAME = $!_UDEV_SYSNAME;
+    }
+    unset $!_UDEV_SYSNAME;
+    if strlen($!_UDEV_DEVNODE) > 0 then {
+        set $!systemd!k!UDEV_DEVNODE = $!_UDEV_DEVNODE;
+    }
+    unset $!_UDEV_DEVNODE;
+    if strlen($!_UDEV_DEVLINK) > 0 then {
+        set $!systemd!k!UDEV_DEVLINK = $!_UDEV_DEVLINK;
+    }
+    unset $!_UDEV_DEVLINK;
+
+    # these fields require special handling
+    if strlen($!level) == 0 then {
+        if strlen($!PRIORITY) > 0 then {
+            set $!level = lookup("prio_to_level", $!PRIORITY);
+        }
+    }
+    unset $!PRIORITY;
+    if strlen($!message) == 0 then {
+        if strlen($!MESSAGE) > 0 then {
+            set $!message = $!MESSAGE;
+        } else {
+            if strlen($!log) > 0 then {
+                set $!message = $!log;
+            }
+        }
+    }
+    unset $!MESSAGE;
```

Comment:
Should we only unset `$!MESSAGE` when we assign it at line 164?  And likewise unset `$!log` at line 167?

Reply:
Neither `MESSAGE` nor `log` are fields in the common data model.  I haven't implemented support for pipeline_metadata - I think these should go into the `original_raw_message` field.

Reply:
@portante we still have a long way to go before rsyslog is a drop in replacement for fluentd - it wasn't my intention to do that in this PR - this PR is only intended to test the basic k8s metadata functionality for $LARGE_CUSTOMER, and as a side effect start the other es and formatting work

- parsing of the JSON embedded in the log/MESSAGE field and assign those to top level fields - in this PR the parsing is done with mmnormalize, but it cannot set those fields as top level fields - see http://lists.adiscon.net/pipermail/rsyslog/2018-March/045207.html
- the rest of fluent-plugin-viaq: stripping empty fields, adding pipeline metadata, eventrouter
- docker audit log parsing
- crio file support
- other parts of the pipeline as found in various bits of fluentd .conf files